grype/match/match.go

```diff
@@ -42,17 +44,56 @@ func (m *Match) Merge(other Match) error {
 		return ErrCannotMerge
 	}
 
+	// there are cases related vulnerabilities are synthetic, for example when
```

Comment:
The changes below here are the main gist of this PR unrelated to the CPE refactoring